### PR TITLE
Great fall refactor

### PR DESCRIPTION
**Pull request by mgale456**
All of the initial setup needed to begin rebuilding the pages and adding new features while following more best-practices are in place.

**Main goal:** Remake `models.py` and refactor the rest of the project to not entirely break. Also add some comments

**More specifically** these changes were made in this branch:

1. Made new model file with new database tables and fields
2. Refactored `admin.py`
3. Refactored initial data migration for the new models
4. Renamed views and urls to match new model names
5. Commented out most of the `views.py` and `consumers.py`, since they will be reworked
6. Refactored urls in templates
7. Renamed templates to match new model names
8. Added all new models to the API

Let me know what you think and feel free to request changes or commit them yourself!

Note that you will need to migrate the models to rebuild your database tables again. Be sure to remove your old docker volume with either `docker-compose down -v` (with running containers) or `docker volumes rm cogs_web` before redoing your migrate with `docker-compose exec web python manage.py migrate`

### CODE REVIEW

**Review by MGhanayem63 — CHANGES_REQUESTED** [review]
Overall almost everything seems to be implemented correctly, here are the things that need to be fixed

Making "Special name" optional for match events
Making sure that the Scoring Event correctly displays
Creating help text for certain fields that may be confusing to understand
Implementing Regex so we can dectect info like phone numbers
Optional if there is time
Implementing the logo of teams and schools using image containers

Otherwise everything else is fine

**Comment by mgale456** [comment]
> Overall almost everything seems to be implemented correctly, here are the things that need to be fixed
> 
> Making "Special name" optional for match events
> Making sure that the Scoring Event correctly displays
> Creating help text for certain fields that may be confusing to understand
> Implementing Regex so we can dectect info like phone numbers
> Optional if there is time
> Implementing the logo of teams and schools using image containers
> 
> Otherwise everything else is fine

1. Made more fields optional
2. Fixed `__str__` display function for `ScoringEvent`
3. Added help text for less straight-forward fields
4. Implemented Regex phone number validation

~~Did not implement logo storing~~ Added basic image storing for school and team logos

These changes require rebuilding the python container and migrating the database again

**Comment by mgale456** [comment]
These initial pull requests should be merged in this order just in case, to ensure no merge issues, since each of the PRs changed a wide variety of files.

1. quick-fixes
2. rest-framework
3. great-fall-refactor

File: django/composeexample/settings.py [hunk 1]
```diff
@@ -143,9 +150,10 @@
 os.path.join(BASE_DIR, "static"),
 ("icons", os.path.join('static/icons')),
 ("custom_css", os.path.join('static/css')),
-
 )
 
+MEDIA_ROOT = os.path.join(BASE_DIR, "static", "logos")
```
Comment (MGhanayem63) [inline django/composeexample/settings.py]: Fix implementation so image can be viewed without error.

**Review by MGhanayem63 — CHANGES_REQUESTED** [review]
Remove commented old lines of code

**Comment by mgale456** [comment]
I made some more updates, this will definitely require rebuilding the Docker images, since there is another container to build, but not deleting and re-migrating your database.

Read the commit comments for what changed. Feel free to ask more questions. 

TLDR for Nginx and Uvicorn:

- Uvicorn is a production grade ASGI server capable of running Django as its backend
- Nginx is a reverse proxy server that allows us to control and route incoming web requests. In our case we are having Nginx serve static files (images, js, css), but pass off the web page requests to Django. We are splitting this by the url paths.

**Review by MGhanayem63 — CHANGES_REQUESTED** [review]
Old comments were removed : )

Images now work :)

Just something I thought about, but I think that if we are going in the direction of using image fields in our database, it would be a good idea to put an image field for contender position objects. The reason being that we have different crests (images) for each position and if MRDC decides to do something similar again another year, then the structure is already there for them.

Although this is most likely work for the scoreboard page, I think it would be a good idea to make some kind of image display somewhere on the page (at least temporarily) to make sure that we can correctly, and without error, display images onto the page even if we can not view them within the admin page.

**Comment by mgale456** [comment]
I'm not sure what errors you're still running into with serving images with Nginx. Let's follow up in Slack for some troubleshooting.

I don't think we should add images to a page as an example, or adding in crests. I think that would be feature creeping this branch. This branch should set it up for success, and ensure the setup is there. It should be easy enough to display an image if we can store and fetch it. Also, adding a field to store the crest is not too much more work and could easily be part of a future branch adding the crests to some page somewhere.

For Nginx and Uvicorn, they should not change anything noticeable other than correctly serving the images stored with the database. They're both backend changes with how we serve the pages/files that what we will be using in production. Previously we were using the dev backend, which is convenient, but not secure or efficient.

**Review by MGhanayem63 — APPROVED** [review]
Everything seems fine, sorry about the mistake I made earlier, everything seems to work fine at least on the front end.

And as for what you said earlier, that makes sense and it should be work for later.

**Review by lpulley — APPROVED** [review]
